types/decimal.go

```diff
@@ -48,7 +50,75 @@ func (d Decimal) UnscaledValue() *big.Int {
 }
 
 // Scale returns the scale of the decimal.
-// The returned value is nonnegative and less or equal to math.MaxInt32.
 func (d Decimal) Scale() int {
 	return int(d.scale)
 }
+
+// Float64 converts the decimal to a float64 and returns it.
+// Note that this conversion can lose information about the precision of the decimal value.
+func (d Decimal) Float64() float64 {
+	f, err := strconv.ParseFloat(d.String(), 64)
+	if err != nil {
+		return 0.0
+	}
```

Comment:
Java BigDecimal implementation uses string parsing method, but I have applied your comment since it gives the same results. See: https://developer.classpath.org/doc/java/math/BigDecimal-source.html#line.1360